pkg/api/routes.go

```diff
@@ -103,40 +109,47 @@ func (rh *RouteHandler) SetupRoutes() {
 	// https://github.com/opencontainers/distribution-spec/blob/main/spec.md#endpoints
 	{
 		prefixedDistSpecRouter.HandleFunc(fmt.Sprintf("/{name:%s}/tags/list", zreg.NameRegexp.String()),
-			applyCORSHeaders(rh.ListTags)).Methods(zcommon.AllowedMethods("GET")...)
+			getUIHeadersHandler(rh.c.Config, http.MethodGet, http.MethodOptions)(
+				applyCORSHeaders(rh.ListTags))).Methods(http.MethodGet, http.MethodOptions)
 		prefixedDistSpecRouter.HandleFunc(fmt.Sprintf("/{name:%s}/manifests/{reference}", zreg.NameRegexp.String()),
-			applyCORSHeaders(rh.CheckManifest)).Methods(zcommon.AllowedMethods("HEAD")...)
+			getUIHeadersHandler(rh.c.Config, http.MethodHead, http.MethodGet, http.MethodOptions)(
+				applyCORSHeaders(rh.CheckManifest))).Methods(http.MethodHead, http.MethodOptions)
 		prefixedDistSpecRouter.HandleFunc(fmt.Sprintf("/{name:%s}/manifests/{reference}", zreg.NameRegexp.String()),
-			applyCORSHeaders(rh.GetManifest)).Methods(zcommon.AllowedMethods("GET")...)
+			getUIHeadersHandler(rh.c.Config, http.MethodHead, http.MethodGet, http.MethodOptions)(
+				applyCORSHeaders(rh.GetManifest))).Methods(http.MethodGet, http.MethodOptions)
```

Comment:
```suggestion
				applyCORSHeaders(rh.GetManifest))).Methods(http.MethodGet)
```
Since line 116 takes care of method OPTIONS.